File: src/core/config/featureFlags.ts
/**
 * Feature flags — управление незавершёнными/экспериментальными функциями.
 *
 * Когда flag = false, функция:
 * - скрыта из UI
 * - при вызове через голос возвращает «Функция пока недоступна»
 *
 * Значения можно переопределить через .env (FEATURE_*)
 * или удалённый конфиг в будущем.
 */

export const FeatureFlags = {
  /** Классификатор денежных купюр (Should-have) */
  banknoteClassifier: true,

  /** Голосовой онбординг при первом запуске (Should-have) */
  onboarding: false,

  /** Описание позиции объектов: лево/центр/право (Could-have, post-MVP) */
  objectPosition: false,

  /** Тест: кнопка на главном экране сразу запускает «Прочитай» */
  testReadOnButtonPress: true,
} as const;

export type FeatureFlagKey = keyof typeof FeatureFlags;

/**
 * Проверяет, включена ли функция.
 * Единая точка входа — удобно для будущей интеграции с remote config.
 */
export function isFeatureEnabled(flag: FeatureFlagKey): boolean {
  return FeatureFlags[flag];
}

File: lib/core/config/feature_flags.dart
/// Feature flags для управления незавершёнными функциями (§4.7 ТЗ).
///
/// Если флаг отключён, соответствующая команда не отображается в UI
/// и возвращает «Функция пока недоступна» при попытке вызова.
///
/// В MVP значения захардкожены; в будущем можно переключить
/// на Remote Config или загрузку из .env.
class FeatureFlags {
  FeatureFlags._();

  /// Классификатор денежных купюр (Should-have, §4.7).
  /// Скрыт за флагом до готовности модели и прохождения тестов.
  static const bool banknoteClassifier = false;

  /// Голосовой туториал при первом запуске (Should-have, §4.8).
  static const bool voiceOnboarding = false;

  /// Описание расположения объектов (Could-have, §4.9).
  static const bool objectPositionDescription = false;
}
